marconi-core/src/Marconi/Core/Experiment/Transformer/WithResume.hs

```diff
@@ -159,71 +167,90 @@ rollbackToAndInsert
+  , MonadIO m
+  , Show (Point event)
+  , Show event
   )
   => Class.IsIndex m event (WithResume indexer)
   where
   index timedEvent indexer =
     let lastPassed' = fromMaybe Class.genesis $ indexer ^. lastPassed
-        invalidHistory =
-          InvalidIndexer
+        invalidHistoryError =
+          ResumingFailed
             "The impossible happened: Indexer history is not compatible with the ledger"
-     in case decidePointStatus (timedEvent ^. point) (indexer ^. lastPoints) of
-          -- We have exhausted the sync list, we're past the resume point and we can index normally
-          ResumingEnded -> Wrapper.indexVia Wrapper.unwrap timedEvent indexer
-          -- the point is before the next point we're waiting
-          OngoingResuming Before -> case indexer ^. lastPassed of
-            -- we haven't process any point yet, we're in a draining phase, so we skip the event
-            Nothing -> pure indexer
-            -- we have processed an event, so the next one should match what we have
-            -- as a consequence, we rollback and then index this new event.
-            -- Here is an example of how this can happen:
-            --   1. you got two blocks, one at slot 200, the next at 220
-            --   2. the indexer stops, while offline the node rollbacks and replace the block at
-            --   slot 220 by a new block, issued at 210
-            --   3. on resume, you correctly received the block 200
-            --   4. then you were expecting a block at slot 220 but received one with a earlier slot
-            --   the only reason is that the node has handled a rollback, we do the same
-            Just p -> rollbackToAndInsert p (Just timedEvent) indexer
-          -- we found our last sync point, so we mark it as passed and we continue draining
-          OngoingResuming Same -> pure $ indexer & resumeConfig %~ drainNextLastPoint
-          -- we found an unexpected next point, so we rollbackTo the previous one stop draining
-          OngoingResuming After -> case indexer ^. lastPassed of
-            Nothing -> throwError invalidHistory
-            -- Here is an example of how this can happen (similar scenario to the 'Before' case):
-            --   1. you got two blocks, one at slot 200, the next at 210
-            --   2. the indexer stops, while offline the node rollbacks and replace the block at
-            --   slot 210 by a new block, issued at 220
-            --   3. on resume, you correctly received the block 200
-            --   4. then you were expecting a block at slot 2'0 but received one with a more recent
-            --   slot the only reason is that the node has handled a rollback, we do the same
-            Just p -> rollbackToAndInsert p (Just timedEvent) indexer
-          -- we found an unexpected next point, so we rollbackTo the previous one stop draining
-          OngoingResuming Fork -> rollbackToAndInsert lastPassed' (Just timedEvent) indexer
+     in do
```

Comment:
why do we have a do here? (I would either remove it or put it on top, to replace the `let .. in` by a simple `let`)

Reply:
Leftover from doing some `print`. Removing it.